src/lib/apexPmd.ts

```diff
@@ -128,24 +128,30 @@ export class ApexPmd {
     const rulesetsArg = this.rulesets.join(',');
 
     const cacheKey = enableCache ? `--cache "${cachePath}"` : '--no-cache';
+    const noProgressBar = '--no-progress';
     const formatKey = `-f csv`;
     const targetPathKey = `-d "${targetPath}"`;
     const rulesetsKey = `-R "${rulesetsArg}"`;
 
-    const pmdKeys = `${formatKey} ${cacheKey} ${targetPathKey} ${rulesetsKey}`;
+    const pmdKeys = `${noProgressBar} ${formatKey} ${cacheKey} ${targetPathKey} ${rulesetsKey}`;
 
     const classPath = [
-      path.join(pmdBinPath, 'lib', '*'),
       path.join(workspaceRootPath, '*'),
       ...additionalClassPaths,
     ].join(CLASSPATH_DELM);
 
-    const javaExc = this.config.jrePath ? path.join(this.config.jrePath, 'bin', 'java') : 'java';
-    const cmd = `${javaExc} -cp "${classPath}" net.sourceforge.pmd.PMD ${pmdKeys}`;
+    let env : NodeJS.ProcessEnv = {};
+    env["CLASSPATH"] = classPath;
+    if (this.config.jrePath) {
+      env["JAVA_HOME"] = this.config.jrePath;
+    }
+
+    const cmd = `${pmdBinPath}/bin/pmd check ${pmdKeys}`;
```

Comment:
we could actually make the part `${pmdBinPath}/bin/pmd check` configurable or even automatically detect, whether we have a PMD 7 or a PMD 6 binary (PMD 6 would use `${pmdBinPath}/bin/run.sh pmd`). That way, we could make that change backwards compatible, if someone wants to use PMD 6 by providing the pmdBinPath preference.